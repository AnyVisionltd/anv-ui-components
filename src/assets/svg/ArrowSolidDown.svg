<svg xmlns="http://www.w3.org/2000/svg" id="prefix__Arrow_solid_down" width="24" height="24" data-name="Arrow solid down" viewBox="0 0 24 24">
    <path id="prefix__Shape" d="M0 0H24V24H0z" style="fill:#02234b;opacity:0"/>
    <path id="prefix__Arrow_solid_down-2" d="M5.247.86a1 1 0 0 1 1.505 0l3.8 4.339A1 1 0 0 1 9.8 6.857H2.2A1 1 0 0 1 1.451 5.2z" data-name="Arrow solid down" transform="rotate(180 9 8.116)"/>
</svg>
