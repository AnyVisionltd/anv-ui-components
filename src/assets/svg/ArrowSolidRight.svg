<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
    <path d="M5.247.86a1 1 0 0 1 1.505 0l3.8 4.339A1 1 0 0 1 9.8 6.857H2.2A1 1 0 0 1 1.451 5.2z" transform="rotate(90 5.088 11.09)"/>
</svg>
