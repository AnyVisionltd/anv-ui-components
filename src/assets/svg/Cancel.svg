<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
    <g id="prefix__Cancel" transform="translate(-82 -8)">
        <path id="prefix__Shape" d="M0 0H16V16H0z" transform="translate(82 8)" style="opacity:0"/>
        <path id="prefix__Path_2223" d="M12.13 11l2.634-2.634a.8.8 0 1 0-1.131-1.131L11 9.868 8.365 7.234a.8.8 0 1 0-1.131 1.131L9.868 11l-2.634 2.633a.8.8 0 1 0 1.131 1.131L11 12.13l2.634 2.634a.8.8 0 1 0 1.131-1.131z" data-name="Path 2223" transform="translate(79.001 5.001)"/>
    </g>
</svg>
