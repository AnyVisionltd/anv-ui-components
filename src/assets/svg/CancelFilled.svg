<svg xmlns="http://www.w3.org/2000/svg" id="prefix__Cancel_X" width="24" height="24" data-name="Cancel X" viewBox="0 0 24 24">
    <path id="prefix__Shape" d="M0 0H24V24H0z" style="opacity:0;fill:#02234b"/>
    <path id="prefix__Subtraction_5" d="M8 16A8 8 0 0 1 2.344 2.343a8 8 0 1 1 11.312 11.31A7.941 7.941 0 0 1 8 16zm0-6.871l2.633 2.634a.841.841 0 0 0 .316.2.806.806 0 0 0 .251.037.791.791 0 0 0 .564-.234A.8.8 0 0 0 12 11.2a.793.793 0 0 0-.234-.564L9.131 8l2.633-2.634a.8.8 0 0 0 0-1.131.8.8 0 0 0-1.131 0L8 6.87 5.365 4.236a.8.8 0 0 0-1.131 0 .8.8 0 0 0 0 1.131L6.869 8l-2.635 2.635A.792.792 0 0 0 4 11.2a.8.8 0 0 0 .8.8.791.791 0 0 0 .564-.234L8 9.131z" data-name="Subtraction 5" transform="translate(4 4)"/>
</svg>
