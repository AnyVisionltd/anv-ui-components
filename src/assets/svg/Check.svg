<svg xmlns="http://www.w3.org/2000/svg" id="prefix__check" width="24" height="24" viewBox="0 0 24 24">
    <path id="prefix__Shape" d="M0 0H24V24H0z" style="opacity:0;fill:#00d0a0"/>
    <path id="prefix__check-2" d="M18.169 7.2a.961.961 0 0 0-1.364 0l-7.157 7.164-3.006-3.016a.982.982 0 1 0-1.364 1.412l3.689 3.689a.961.961 0 0 0 1.364 0l7.838-7.839a.961.961 0 0 0 0-1.412z" data-name="check" transform="translate(.272 -.163)" />
</svg>
