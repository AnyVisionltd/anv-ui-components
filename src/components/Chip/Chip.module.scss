@import "~@anyvision/style-guide";
@import "../../styles/mixin/ripple";

.chip {
  outline: none;
  box-sizing: border-box;
  position: relative;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: av-size(sz-32);
  padding: av-space(micro);
  border-radius: av-radius(round);
  background-color: av-color(primary, av-opacity(divider));
  color: av-color(primary);
  fill: currentColor;

  &.focusable {
    @include av-ripple();

    &:focus {
      background-color: av-color(primary, av-opacity(disabled));
    }
  }

  &.isFocused {
    background-color: av-color(primary, av-opacity(disabled));
  }

  &.clickable {
    cursor: pointer;
  }

  &.disabled {
    opacity: av-opacity(disabled);
  }

  .leadingIcon,
  .trailingIcon {
    svg,
    img {
      display: block;
    }

    img {
      border-radius: 100%;
    }
  }

  .leadingIcon {
    height: av-size(sz-24);
    width: av-size(sz-24);

    img {
      border-radius: 100%;
    }
  }

  .trailingIcon {
    position: relative;
    background: none;
    border: none;
    padding: 0;
    color: inherit;
    height: av-size(sz-16);
    width: av-size(sz-16);
    display: flex;
    justify-content: center;
    align-items: center;
    margin: av-space(micro);
    margin-inline-start: 0;
    z-index: 1; // To overcome ripple layer drawing above the icon
    cursor: default; // To reset button's pointer cursor - trailing icon isn't clickable by default

    &.trailingIconClickable {
      cursor: pointer;

      &:before {
        border-radius: 100%;
        position: absolute;
        height: 100%;
        width: 100%;
        display: block;
        content: '';
        background-color: currentColor;
        opacity: 0;
      }

      &:hover {
        &:before {
          opacity: av-opacity(disabled);
        }
      }
    }
  }

  .label {
    padding: 0 av-space(tiny);
  }

  > button {
    outline: none;
  }
}
