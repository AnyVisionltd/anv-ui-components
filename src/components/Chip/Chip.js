import React, { useEffect } from 'react'
import classNames from 'classnames'
import propTypes from 'prop-types'
import { ReactComponent as CloseIcon } from '../../assets/svg/Cancel.svg'
import styles from './Chip.module.scss'

const Chip = ({
  leadingIcon,
  className,
  trailingIcon,
  disabled,
  deletable,
  label,
  isFocused,
  onClick,
  onTrailingIconClick,
  onKeyUp,
  ...otherProps
}) => {
  const chipRef = React.createRef()
  const clickable = !disabled && !!onClick
  const focusable = clickable || deletable

  useEffect(() => {
    if (isFocused) {
      chipRef.current.focus()
    } else {
      chipRef.current.blur()
    }
  }, [isFocused])

  const classes = classNames(
    styles.chip,
    focusable && styles.focusable,
    clickable && styles.clickable,
    disabled && styles.disabled,
    className,
  )

  const isDeleteKeyboardEvent = event => deletable && ['Backspace', 'Delete'].includes(event.key)

  const handleKeyUp = event => {
    if (event.currentTarget === event.target) {
      if (isDeleteKeyboardEvent(event) && onTrailingIconClick) {
        onTrailingIconClick(event)
      } else if (event.key === 'Enter' && chipRef.current && clickable) {
        onClick(event)
      } else if (event.key === 'Escape' && chipRef.current) {
        chipRef.current.blur()
      }
    }

    return onKeyUp && onKeyUp(event)
  }

  const onClickHandler = event => {
    if (!clickable) {
      return
    }

    onClick(event)
  }

  const onTrailingIconClickHandler = event => {
    if (!onTrailingIconClick) {
      return
    }
    event.stopPropagation()
    // When clicked using mouse, we would expect the chip to not be focused anymore.
    chipRef.current.blur()
    onTrailingIconClick(event)
  }

  const renderLeadingIcon = () => leadingIcon && (
    <div className={ styles.leadingIcon }>
      { leadingIcon }
    </div>
  )

  const renderTrailingIcon = () => {
    let displayedTrailingIcon = trailingIcon
    if (!displayedTrailingIcon) {
      if (onTrailingIconClick && deletable) {
        displayedTrailingIcon = <CloseIcon />
      } else {
        return null
      }
    }

    const trailingIconClasses = classNames(
      styles.trailingIcon,
      onTrailingIconClick && styles.trailingIconClickable,
    )

    return (
      <button
        className={ trailingIconClasses }
        onClick={ onTrailingIconClickHandler }
      >
        { displayedTrailingIcon }
      </button>
    )
  }

  const renderLabel = () => <span className={ styles.label }>{ label }</span>

  return (
    <div
      role="button"
      aria-disabled={ disabled ? true : undefined }
      tabIndex={ focusable ? 0 : undefined }
      className={ classes }
      onClick={ onClickHandler }
      onKeyUp={ handleKeyUp }
      ref={ chipRef }
      { ...otherProps }
    >
      { renderLeadingIcon() }
      { renderLabel() }
      { renderTrailingIcon() }
    </div>
  )
}

Chip.defaultProps = {
  disabled: false,
}

Chip.propTypes = {
  /** The icon of the chip. */
  leadingIcon: propTypes.element,
  /** For css customization. */
  className: propTypes.string,
  /** If <code>true</code>, the chip will be disabled. */
  disabled: propTypes.bool,
  /** The label/content of the chip */
  label: propTypes.node,
  /** Make the chip focused */
  isFocused: propTypes.bool,
  /**
   * Callback function fired when clicked.
   * Passing a function will make the chip look clickable (by hover effect, mouse pointer, focus).
   */
  onClick: propTypes.func,
  /**
   * The trailing icon for the chip.
   * If not set, <u>but</u> <code>onTrailingIconClick</code> and <code>deletable</code> are passed,
   * this button will default to closeIcon
   */
  trailingIcon: propTypes.element,
  /**
   * If enabled, the chip will handle backspace or delete keyboard actions by calling
   * <code>onTrailingIconClick</code> handler when focused.
   */
  deletable: propTypes.bool,
  /**
   * Callback function when trailing icon clicked.
   */
  onTrailingIconClick: propTypes.func,
  onKeyUp: propTypes.func,
}

export default Chip
