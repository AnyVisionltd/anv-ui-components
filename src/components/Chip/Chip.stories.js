import React from 'react'
import { boolean, text } from '@storybook/addon-knobs'
import Chip from './Chip'
import { SunBrightens } from '@anyvision/anv-icons'
import { ReactComponent as CheckedIcon } from '../../assets/svg/Checked.svg'
import { centerDecorator } from '../../utils/storybook/decorators'
import styles from '../../storybook/index.module.scss'

export default {
  title: 'Content/Chip',
  component: Chip,
  decorators: [centerDecorator],
}

export const Default = () => (
  <Chip
    label='Chocolate'
    leadingIcon={<SunBrightens />}
    onClick={() => {}}
    onTrailingIconClick={() => {}}
    deletable
  />
)

export const Variants = () => (
  <>
    <Chip label='Simple' className={styles.microMargin} />
    <Chip label='Clickable' onClick={() => {}} className={styles.microMargin} />
    <Chip label='Disabled' className={styles.microMargin} disabled />
    <Chip
      label='Leading Icon'
      className={styles.microMargin}
      leadingIcon={<SunBrightens />}
    />
    <Chip
      label='Trailing Icon'
      className={styles.microMargin}
      trailingIcon={<CheckedIcon />}
    />
    <Chip
      label='Both Icons'
      className={styles.microMargin}
      leadingIcon={<SunBrightens />}
      trailingIcon={<CheckedIcon />}
    />
  </>
)

export const playground = () => {
  const leadingIcon = boolean('Display leading icon?', false) ? (
    <SunBrightens />
  ) : null
  const isDeletable = boolean(
    'Will clicking on trailing icon end with deleting the chip?',
    true,
  )
  const trailingIcon = boolean('Display custom trailing icon?', false) ? (
    <CheckedIcon />
  ) : null
  return (
    <>
      <Chip
        label={text('label', 'Chip')}
        className={styles.microMargin}
        leadingIcon={leadingIcon}
        trailingIcon={trailingIcon}
        onClick={() => alert('Chip was clicked!')} // eslint-disable-line no-alert
        onTrailingIconClick={() =>
          alert(
            isDeletable ? 'Yep, will delete' : 'Trailing icon click handler!',
          )
        } // eslint-disable-line no-alert
        deletable={isDeletable}
      />
      {isDeletable && (
        <>
          <small className={styles.tinyText}>
            Try focusing the chip using your keyboard, then clicking on
            backspace!
          </small>
        </>
      )}
    </>
  )
}
