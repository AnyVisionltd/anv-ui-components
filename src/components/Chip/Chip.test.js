import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { ReactComponent as SvgIcon } from '../../../jest/assets/svgIcon.svg'
import Chip from './Chip'

const chipText = 'Chip text'
describe('<Chip />', () => {
  it('should fire click handler if \'onClick\' prop is set', () => {
    const handleClick = jest.fn()
    const { getByRole } = render(
      <Chip
        label={ chipText }
        onClick={ handleClick }
      />,
    )
    const node = getByRole('button')
    fireEvent.click(node)
    expect(handleClick).toBeCalled()
  })

  it('should not be clickable when disabled', () => {
    const handleClick = jest.fn()
    const { getByRole } = render(
      <Chip
        label={ chipText }
        disabled
        onClick={ handleClick }
      />,
    )
    const node = getByRole('button')
    fireEvent.click(node)
    expect(handleClick).not.toBeCalled()
  })

  it('should be clickable by pressing return key while focused', () => {
    const handleClick = jest.fn()
    const { getByRole } = render(
      <Chip
        label={ chipText }
        onClick={ handleClick }
      />,
    )
    const node = getByRole('button')
    fireEvent.focus(node)
    fireEvent.keyUp(node, { key: 'Enter' })
    expect(handleClick).toBeCalled()
  })

  it('should call \'on trailing icon click\' when set as deletable and pressed on backspace key while focused', () => {
    const handleClick = jest.fn()
    const { getAllByRole } = render(
      <Chip
        label={ chipText }
        deletable
        onTrailingIconClick={ handleClick }
      />,
    )
    const [node] = getAllByRole('button')
    fireEvent.focus(node)
    fireEvent.keyUp(node, { key: 'Backspace' })
    expect(handleClick).toBeCalled()
  })

  it('should call \'on trailing icon click\' when set as deletable and pressed on delete key while focused', () => {
    const handleClick = jest.fn()
    const { getAllByRole } = render(
      <Chip
        label={ chipText }
        deletable
        onTrailingIconClick={ handleClick }
      />,
    )
    const [node] = getAllByRole('button')
    fireEvent.focus(node)
    fireEvent.keyUp(node, { key: 'Delete' })
    expect(handleClick).toBeCalled()
  })

  it('should call \'on trailing icon click\' when trailing icon is clicked', () => {
    const handleClick = jest.fn()
    const { getAllByRole } = render(
      <Chip
        label={ chipText }
        onTrailingIconClick={ handleClick }
        trailingIcon={ <SvgIcon /> }
      />,
    )
    const [, node] = getAllByRole('button')
    fireEvent.click(node)
    expect(handleClick).toBeCalled()
  })
})
