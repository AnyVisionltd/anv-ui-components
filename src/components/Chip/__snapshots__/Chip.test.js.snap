// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Button Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill"
      disabled={false}
      onClick={[Function]}
    >
      Text
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Disable 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill"
        disabled={false}
        onClick={[Function]}
      >
        Enable
      </button>
      <button
        className="button large fill"
        disabled={true}
        onClick={[Function]}
      >
        Disabled
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Button Play Ground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill"
      disabled={false}
      onClick={[Function]}
    >
      Button Text
    </button>
  </div>
</div>
`;

exports[`Storyshots Button Sizes 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill"
        disabled={false}
        onClick={[Function]}
      >
        Large
      </button>
      <button
        className="button small fill"
        disabled={false}
        onClick={[Function]}
      >
        small
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Button Variants 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill"
        disabled={false}
        onClick={[Function]}
      >
        fill
      </button>
      <button
        className="button large outline"
        disabled={false}
        onClick={[Function]}
      >
        outline
      </button>
      <button
        className="button large ghost"
        disabled={false}
        onClick={[Function]}
      >
        ghost
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Button With Class Name 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill successBackgroundColor"
        disabled={false}
        onClick={[Function]}
      >
        Text
      </button>
      <button
        className="button large outline alertColor"
        disabled={false}
        onClick={[Function]}
      >
        Text
      </button>
      <button
        className="button large ghost errorColor"
        disabled={false}
        onClick={[Function]}
      >
        Text
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Button With Icon 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill"
      disabled={false}
      onClick={[Function]}
    >
      <span
        className="leadingIcon"
      >
        <svg />
      </span>
      leading
    </button>
  </div>
</div>
`;

exports[`Storyshots Checkbox Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox enabled"
      htmlFor="checkbox-example"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        id="checkbox-example"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
    <label
      className="checkboxLabel"
      htmlFor="checkbox-example"
    >
      Check me out!
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox Playground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox enabled"
      htmlFor="checkbox-playground"
    >
      <input
        checked={false}
        className="parent"
        color="primary"
        disabled={false}
        id="checkbox-playground"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
    <label
      className="checkboxLabel"
      htmlFor="checkbox-playground"
    >
      Hit me!
    </label>
  </div>
</div>
`;

exports[`Storyshots Checkbox States 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <label
      className="checkbox checked enabled microMargin"
    >
      <input
        checked={true}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      >
        <svg />
      </span>
    </label>
    <label
      className="checkbox enabled microMargin"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
    <label
      className="checkbox checked disabled microMargin"
    >
      <input
        checked={true}
        className="parent"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      >
        <svg />
      </span>
    </label>
    <label
      className="checkbox indeterminate enabled microMargin"
    >
      <input
        checked={false}
        className="parent"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      >
        <svg />
      </span>
    </label>
    <label
      className="checkbox disabled microMargin"
    >
      <input
        checked={false}
        className="parent"
        disabled={true}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="iconContainer"
      />
    </label>
  </div>
</div>
`;

exports[`Storyshots Chip Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="chip focusable clickable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="leadingIcon"
      >
        <svg />
      </div>
      <span
        className="label"
      >
        Chocolate
      </span>
      <button
        className="trailingIcon trailingIconClickable"
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Chip Playground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="chip focusable clickable microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="label"
      >
        Chip
      </span>
      <button
        className="trailingIcon trailingIconClickable"
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
    <small
      className="tinyText"
    >
      Try focusing the chip using your keyboard, then clicking on backspace!
    </small>
  </div>
</div>
`;

exports[`Storyshots Chip Variants 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="chip microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      <span
        className="label"
      >
        Simple
      </span>
    </div>
    <div
      className="chip focusable clickable microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="label"
      >
        Clickable
      </span>
    </div>
    <div
      aria-disabled={true}
      className="chip disabled microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      <span
        className="label"
      >
        Disabled
      </span>
    </div>
    <div
      className="chip microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      <div
        className="leadingIcon"
      >
        <svg />
      </div>
      <span
        className="label"
      >
        Leading Icon
      </span>
    </div>
    <div
      className="chip microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      <span
        className="label"
      >
        Trailing Icon
      </span>
      <button
        className="trailingIcon"
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
    <div
      className="chip microMargin"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      <div
        className="leadingIcon"
      >
        <svg />
      </div>
      <span
        className="label"
      >
        Both Icons
      </span>
      <button
        className="trailingIcon"
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots IconButton Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill iconButton small fill"
      disabled={false}
      onClick={[Function]}
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Disable 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill iconButton small fill"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
      <button
        className="button large fill iconButton small fill"
        disabled={true}
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots IconButton Play Ground 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill iconButton small fill"
      disabled={false}
      onClick={[Function]}
    >
      <svg />
    </button>
  </div>
</div>
`;

exports[`Storyshots IconButton Sizes 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill iconButton small fill"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
      <button
        className="button large fill iconButton large fill"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots IconButton Variants 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill iconButton small fill"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
      <button
        className="button large outline iconButton small outline"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
      <button
        className="button large ghost iconButton small ghost"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots IconButton With Class Name 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill iconButton small fill successBackgroundColor"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
      <button
        className="button large outline iconButton small outline alertColor"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
      <button
        className="button large ghost iconButton small ghost errorColor"
        disabled={false}
        onClick={[Function]}
      >
        <svg />
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Snackbar Action Text 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill"
      disabled={false}
      onClick={[Function]}
    >
      Open
    </button>
  </div>
</div>
`;

exports[`Storyshots Snackbar Close Icon 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <div
      className="marginFlexContainer"
    >
      <button
        className="button large fill"
        disabled={false}
        onClick={[Function]}
      >
        Default Icon
      </button>
      <button
        className="button large fill"
        disabled={false}
        onClick={[Function]}
      >
        Custom Icon
      </button>
      <button
        className="button large fill"
        disabled={false}
        onClick={[Function]}
      >
        Without Icon
      </button>
    </div>
  </div>
</div>
`;

exports[`Storyshots Snackbar Default 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill"
      disabled={false}
      onClick={[Function]}
    >
      Open
    </button>
  </div>
</div>
`;

exports[`Storyshots Snackbar Leading Icon 1`] = `
<div
  className="story-wrapper"
>
  <div
    className="center"
  >
    <button
      className="button large fill"
      disabled={false}
      onClick={[Function]}
    >
      Open
    </button>
  </div>
</div>
`;
